dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java

```diff
@@ -4530,6 +4522,150 @@ public void contentOnlyWithExpireFieldTest() throws Exception{
+
+        contentletA = new Contentlet();
+        contentletA.setStructureInode(contentType.inode());
+        contentletA.setLanguageId(languageAPI.getDefaultLanguage().getId());
+        contentletA.setStringProperty(textField.variable(), "A");
+        contentletA.setBinary(binaryField, tempTestFile);
+        contentletA.setIndexPolicy(IndexPolicy.FORCE);
+        contentletA.setIndexPolicyDependencies(IndexPolicy.FORCE);
+        contentletA = contentletAPI.checkin(contentletA, user, false);
+    }
+
+    /**
+     * This test will:
+     * --- Create a content type called "Nested".
+     * --- Add  1 Text field called Title
+     * --- Add  1 Binary field called File
+     * --- Set an application/* as a {@link com.dotcms.contenttype.model.field.BinaryField#ALLOWED_FILE_TYPES}
+     * --- Upload a wrong file
+     * --- expecteds DotContentletValidationException
+     *
+     */
+    @Test (expected = DotContentletValidationException.class)
+    public void test_validateContentlet_wrong_mime_type_expect_DotContentletValidationException() throws Exception {
+
+        ContentType contentType = null;
+        com.dotcms.contenttype.model.field.Field textField   = null;
+        com.dotcms.contenttype.model.field.Field binaryField = null;
+
+        Contentlet contentletA = null;
+
+            // Create Content Type.
+            contentType = ContentTypeBuilder.builder(BaseContentType.CONTENT.immutableClass())
+                    .description("Nested" + System.currentTimeMillis())
+                    .host(defaultHost.getIdentifier())
+                    .name("Nested" + System.currentTimeMillis())
+                    .owner("owner")
+                    .variable("nested" + System.currentTimeMillis())
+                    .build();
+
+            contentType = contentTypeAPI.save(contentType);
+
+            // Save Fields. 1. Text
+            // Creating Text Field: Title.
+            textField = ImmutableTextField.builder()
+                    .name("Title")
+                    .variable("title")
+                    .contentTypeId(contentType.id())
+                    .dataType(DataTypes.TEXT)
+                    .build();
+
+            textField = fieldAPI.save(textField, user);
+
+            // Save Fields. 1. Binary
+            // Creating Text Field: File.
+            binaryField = ImmutableBinaryField.builder()
+                    .name("file")
+                    .variable("file")
+                    .contentTypeId(contentType.id())
+                    .build();
+
+
+            binaryField = fieldAPI.save(binaryField, user);
+
+            FieldVariable maxLength      = ImmutableFieldVariable.builder().key(BinaryField.MAX_FILE_LENGTH).value("10").fieldId(binaryField.id()).build();
+            FieldVariable allowFileTypes = ImmutableFieldVariable.builder().key(BinaryField.ALLOWED_FILE_TYPES).value("application/*").fieldId(binaryField.id()).build();
```

Comment:
I might be wrong but I think the test case validates are exactly the same that `test_validateContentlet_wrong_size_expect_DotContentletValidationException` . MAX_FILE_LENGTH and ALLOWED_FILE_TYPES have the same value of the previous test

Reply:
@nollymar is not the same since the length of the temp file is within the valid length (10). See below:
`FileUtils.writeStringToFile(tempTestFile, "Test");`

Reply:
Almost the issue is doing two validations:
1) max length for a file
2) mime types accepted

So one test for each